bundle/src/main/java/com/adobe/acs/commons/fam/impl/ActionManagerImpl.java

```diff
@@ -131,25 +147,26 @@ public void accept(ResourceResolver r) throws Exception {
                     LOG.error("Repository exception processing query "+queryStatement, ex);
                 }
             }
-        }, false);
+        });
         return tasksAdded.get();
     }
 
     @Override
     public void addCleanupTask() {
-        for (int i = 0; i < resolverCount; i++) {
-            deferredWithResolver(new Consumer<ResourceResolver>() {
-                @Override
-                public void accept(ResourceResolver t) {
-                    try {
-                        t.commit();
-                        t.close();
-                    } catch (PersistenceException ex) {
-                        LOG.error("Persistence exception closing session pool", ex);
+        Runnable r = new Runnable() {
+            @Override
+            public void run() {
+                while (!isComplete()) {
+                    try {                    
+                        Thread.sleep(100);
```

Comment:
100ms?  Nothing terribly significant, I just want to make sure that we're not in a tight wait-loop while other threads complete work and waiting several seconds for that to happen is almost certain.